lib/elastic_record/as_document.rb

```diff
@@ -0,0 +1,50 @@
+module ElasticRecord
+  module AsDocument
+    def as_search_document
+      json = {}
+
+      doctype.mapping[:properties].each do |field, mapping|
+        value = elastic_search_value field, mapping
+
+        unless value.nil?
+          json[field] = value
+        end
+      end
+
+      json
+    end
```

Comment:
Can use `each_with_object`. Original version of this was trying to avoid rails dependencies. The gem is now rails dependent, and each_with_object is in ruby.